ferrocene/ci/scripts/require-uv-shebang.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env -S uv run
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Ferrocene Authors
+
+# /// script
+# requires_python = ">= 3.12"
+# dependencies = []
+# ///
+
+from pathlib import Path
+import os
+import subprocess
+import sys
+
+
+# All files in one of the paths included withing CHECK_PATHS but not in EXCLUDE_PATHS will be
+# checked. Only exclude paths if they are external subtrees not managed by us.
+CHECK_PATHS = ["ferrocene", ".circleci", ".github"]
+EXCLUDE_PATHS = ["ferrocene/library/libc"]
+
+UV_SHEBANG = b"#!/usr/bin/env -S uv"
+
+
+def main():
+    cmd = subprocess.run(
+        ["git", "ls-tree", "HEAD", "--name-only", "-r", "-z"],
+        stdout=subprocess.PIPE,
+        check=True,
+        text=True,
+    )
+
+    # The -z flag changes the output to separate file names with a null byte, preventing files with
+    # spaces or newlines in their names from messing things up.
+    has_errors = False
+    for file in cmd.stdout.split("\0"):
+        file = Path(file)
+
+        def error(message):
+            nonlocal has_errors
```

Comment:
why `nonlocal`?

Reply:
Because it needs to change the variable defined in the parent scope.